src/rest.rs

```diff
@@ -0,0 +1,115 @@
+use crate::OKEx;
+use serde::{Deserialize, Deserializer, Serialize};
+use std::str::FromStr;
+use std::fmt::Display;
+use std::collections::HashMap;
+
+const API_HOST: &'static str = "https://www.okex.com/api/spot/v3/";
+
+pub struct RestClient {
+    client: reqwest::Client,
+    api_key: String,
+    secret: String,
+}
+
+impl OKEx for RestClient {
+    fn new() -> Self {
+        let timeout = std::time::Duration::from_secs(5);
+        // TODO: Will this just panic if there is an error building the client?
+        // Is that alright? Or should the use be able to handle to error.
+        let client = reqwest::ClientBuilder::new()
+            .timeout(timeout).build().unwrap();
```

Comment:
You could return a `Result` (from `std::result`, see [here](https://doc.rust-lang.org/std/result/)) from the `new` method and unwrap this error nicely by calling `build` with a question mark to propagate the error:

```rust
let client = reqwest::ClientBuilder::new()
    .timeout(timeout).build()?;
```

This would obviously mean the signature in the `OKEx` trait would need to change.

Reply:
I agree, and just to add that yes `.unwrap()` will panic if there's nothing there.